streamingccl: add a builtin function that retrives stream ingestion statitics

gh-casper opened the pull request:
Support crdb_internal.stream_ingestion_stats(ingestion_job_id) to
running progress for a stream ingestion job.

Release note (sql change): Support crdb_internal.stream_ingestion_stats
(ingestion_job_id) to running progress for a stream ingestion job.

Release Justification: Cat 4.

Closes: https://github.com/cockroachdb/cockroach/issues/57430

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/79748)
<!-- Reviewable:end -->

miretskiy commented on pkg/ccl/streamingccl/streamproducer/stream_lifetime.go:
```diff
@@ -289,3 +290,30 @@ func completeReplicationStream(
+	evalCtx *tree.EvalContext, txn *kv.Txn, ingestionJobID jobspb.JobID,
+) (*streampb.StreamIngestionStats, error) {
+	registry := evalCtx.Planner.ExecutorConfig().(*sql.ExecutorConfig).JobRegistry
+	j, err := registry.LoadJobWithTxn(evalCtx.Ctx(), ingestionJobID, txn)
```
Do we need to use withTxn variety?

otan commented on pkg/sql/sem/builtins/replication_builtins.go:
```diff
@@ -70,6 +73,45 @@ var replicationBuiltins = map[string]builtinDefinition{
+			},
+
+			Info: "This function can be used on the ingestion side to get a statistics summary in json format " +
+				"of a stream ingestion job",
```
nit: full stop at the end here

otan reviewed (approved):
nit: commit title / pr title misspelling of statistics

gh-casper committed:
streamingccl: add a builtin function that retrives stream
ingestion statitics

Support crdb_internal.Support crdb_internal.
stream_ingestion_stats(ingestion_job_id) to
running progress for a stream ingestion job.

Release note (sql change): Support crdb_internal.stream_ingestion_stats
(ingestion_job_id) to running progress for a stream ingestion job.

gh-casper commented:
bors r+

gh-casper commented:
bors r+
